python/ray/serve/scripts.py

```diff
@@ -232,49 +234,170 @@ def start(
     )
 
 
+def _generate_config_from_file_or_import_path(
+    config_or_import_path: str,
+    *,
+    name: Optional[str],
+    arguments: Dict[str, str],
+    runtime_env: Optional[Dict[str, Any]],
+) -> ServeDeploySchema:
+    """Generates a deployable config schema for the passed application(s)."""
+    if pathlib.Path(config_or_import_path).is_file():
```

Comment:
This is quite an edge case, but this could fail if the import path uses an extension at the end, e.g. `my_file.txt`. Can we also check that the path ends in `.yaml`?

Reply:
`is_file()` checks if the file exists on disk